Source/Core/VideoCommon/HiresTextures.cpp

```diff
@@ -135,25 +145,74 @@ void HiresTexture::Update()
       }
     }
 
-    s_textureCacheAbortLoading.Clear();
+    s_textureCacheAbortPrefetching.Clear();
     s_prefetcher = std::thread(Prefetch);
   }
 }
 
 void HiresTexture::Prefetch()
 {
+  // prefetch() is running in a tight loop, and it's using that to allow "at least" 1 Search()
+  // run per cycle
   Common::SetCurrentThreadName("Prefetcher");
 
-  size_t size_sum = 0;
-  size_t sys_mem = Common::MemPhysical();
-  size_t recommended_min_mem = 2 * size_t(1024 * 1024 * 1024);
-  // keep 2GB memory for system stability if system RAM is 4GB+ - use half of memory in other cases
-  size_t max_mem =
-      (sys_mem / 2 < recommended_min_mem) ? (sys_mem / 2) : (sys_mem - recommended_min_mem);
-  u32 starttime = Common::Timer::GetTimeMs();
-  for (const auto& entry : s_textureMap)
+  double prefetch_start_time_total = 0;
+  double prefetch_stop_time_total = 0;
+  constexpr double CULLED_PERCENT_DEFAULT = 30.0;
+  constexpr double CULLED_PERCENT_LOW_BOUND = 5.0;
+  constexpr double CULLED_PERCENT_HIGH_BOUND = 90.0;
+
+  // Overridable by HiresTexOverridePrefetchCulledPercent INI setting
+  double culled_percent = CULLED_PERCENT_DEFAULT;
+
+  if (Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT) != CULLED_PERCENT_DEFAULT)
   {
-    const std::string& base_filename = entry.first;
+    if (Config::Get(Config::GFX_HIRESTEX_OVERRIDE_PREFETCH_CULLED_PERCENT) <
+        CULLED_PERCENT_LOW_BOUND)
+    {
+      culled_percent = CULLED_PERCENT_LOW_BOUND;
+      WARN_LOG(VIDEO, "Prefetcher: Mem limit cull percentage exceeds the minimum of %.2f %%. "
```

Comment:
Do we have a logging policy in terms of out-of-range values? IMO, it's better to specify the "range" of a value in the config declaration and validate it there, rather than littering everywhere which uses these variables with checks and messages.

Reply:
Someone else also suggested breaking off the custom textures logging from the general VIDEO, if that helps with less cluter then it's an easy fix and 2 flies with one stone so I don't have to redo this whole thing.